### PR TITLE
Add `[CustomAssertion]` attribute to extensibility docs

### PR DESCRIPTION
**Pull request by eranikid**
There are (at least) two pages in the docs, that describe creating custom assertion methods - [Introduction](https://fluentassertions.com/introduction#subject-identification) and [Extensibility](https://fluentassertions.com/extensibility/#building-your-own-extensions)

`[CustomAssertion]` attribute is only mentioned in _Introduction_, along with a description of why including it is important, whereas in _Extensibility_ there is no mention of that attribute, and people googling https://www.google.com/search?q=fluent+assertions+how+to+create+custom+assertion will more likely land on _Extensibility_ rather than _Introduction_

This leads to confusion.

This PR adds a mention of `[CustomAssertion]` attribute to _Extensibility_ page

## IMPORTANT 

* [x] If the PR touches the public API, the changes have been approved in a separate issue with the "api-approved" label.
* [x] The code complies with the [Coding Guidelines for C#](https://www.csharpcodingguidelines.com/).
* [x] The changes are covered by unit tests which follow the Arrange-Act-Assert syntax and the naming conventions such as is used [in these tests](../tree/develop/Tests/FluentAssertions.Equivalency.Specs/MemberMatchingSpecs.cs#L51-L430).
* [x] If the PR adds a feature or fixes a bug, please update [the release notes](../tree/develop/docs/_pages/releases.md) with a functional description that explains what the change means to consumers of this library, which are published on the [website](https://fluentassertions.com/releases).
* [x] If the PR changes the public API the changes needs to be included by running [AcceptApiChanges.ps1](../tree/develop/AcceptApiChanges.ps1) or [AcceptApiChanges.sh](../tree/develop/AcceptApiChanges.sh).
* [x] If the PR affects [the documentation](../tree/develop/docs/_pages), please include your changes in this pull request so the documentation will appear on the [website](https://www.fluentassertions.com/introduction).
    * [ ] Please also run `./build.sh --target spellcheck` or `.\build.ps1 --target spellcheck` before pushing and check the good outcome

### CODE REVIEW

**Comment by eranikid** [comment]
I am unfortunately getting a 50 seconds timeout running `./build.sh --target spellcheck`, but I guess there is not much to spellcheck

**Comment by ITaluone** [comment]
You probably have a slow internet connection, so the download of node.js takes more than 50s.

You can increase the timeout locally the first time here: https://github.com/fluentassertions/fluentassertions/blob/a66fe37e587e28bca590c46bbcd8c152e707e43a/Build/CustomNpmTasks.cs#L84 to whatever value.

(BTW: And I thought I have a slow internet connection at home 🙈 )

File: docs/_pages/extensibility.md [hunk 1]
```diff
@@ -36,6 +36,7 @@ public class DirectoryInfoAssertions :
 
     protected override string Identifier => "directory";
 
+    [CustomAssertion]
```
Comment (dennisdoomen) [inline docs/_pages/extensibility.md:39]: 🔧 I think we should at least explain what this attribute does in the text following the example.

**Review by dennisdoomen — CHANGES_REQUESTED** [review]

**Comment by IT-VBFK** [comment]
@eranikid Rebase onto current develop and force push should fix the pipeline..

File: docs/_pages/extensibility.md (hunk 1 above)
Comment (eranikid) [reply docs/_pages/extensibility.md:39]: @dennisdoomen Do you think it is a good idea to duplicate the explanation fron Introduction? https://github.com/fluentassertions/fluentassertions/blob/180c1a33aa41538674fad623c37af4bb20c586ab/docs/_pages/introduction.md?plain=1#L142

Comment (dennisdoomen) [reply docs/_pages/extensibility.md:39]: Just a summary of why it is needed.

Comment (eranikid) [reply docs/_pages/extensibility.md:39]: @dennisdoomen 

This good https://github.com/fluentassertions/fluentassertions/pull/2685/commits/9bebe76c4c7ce89ffed89c33835842ff5def68ef? Or is it too vague and summarized, and I should elaborate more on in what way test fail messages are more meaningful with `[CustomAssertion]` than without it?

**Comment by eranikid** [comment]
@jnyrup I'm seeing your name on the build that failed. Do you have any guess why does it say `Member 'GitVersion' is required to be not null or empty` in the error message?

**Comment by ITaluone** [comment]
Just rebase onto current develop and force push

**Comment by eranikid** [comment]
@ITaluone I do not understand how can I rebase my branch onto develop, when my branch **is** the develop in my fork :D

I've pressed the "Sync fork" button, and it created a merge commit with changes from upstream, that's probably not fixing the issue, right?

**Comment by dennisdoomen** [comment]
> I've pressed the "Sync fork" button, and it created a merge commit with changes from upstream, that's probably not fixing the issue, right?

You need to use a proper Git GUI or understand the command-line tools for rebasing. Maybe read-up on this [here](https://git-scm.com/book/en/v2/Git-Branching-Rebasing)?

**Comment by ITaluone** [comment]
> I do not understand how can I rebase my branch onto develop, when my branch is the develop in my fork :D

This is not the right way, to contribute to a project. The develop in your fork is only to keep in sync with the upstream repo. Every other feature/fix goes into a separate branch in your fork.

**Comment by eranikid** [comment]
> This is not the right way, to contribute to a project. The develop branch in your fork is only to keep in sync with the upstream develop branch (and nothing else). Every other feature/fix goes into a separate branch in your fork.

Oh I see, I thought branch choice in fork does not matter, I'll try to cherry-pick fix into a separate branch

**Comment by dennisdoomen** [comment]
> Oh I see, I thought branch choice in fork does not matter, I'll try to cherry-pick fix into a separate branch

It's not, but from our perspective it doesn't matter which branch you use, as long as you keep the changes clean by rebasing on _our_ `develop`.

**Comment by ITaluone** [comment]
> It's not, but from our perspective it doesn't matter which branch you use, as long as you keep the changes clean by rebasing on our develop.

That's true as well, but you definitely have less troubles keeping your fork-develop clean..

**Pull request closed without merging**